File: src/class10/EqualsOverloadingOverrideToString/exercise1/java/Pessoa.java
package class10.equalsoverloadingoverridetostring.exercise1.java;

import java.util.Objects;

public class Pessoa {
    private String nome;
    private Integer rg;

    public Pessoa(String nome, Integer rg) {
        this.nome = nome;
        this.rg = rg;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public Integer getRg() {
        return rg;
    }

    public void setRg(Integer rg) {
        this.rg = rg;
    }

    //Equals padrão que compara o objeto inteiro
//    @Override
//    public boolean equals(Object o) {
//        if (this == o) return true;
//        if (o == null || getClass() != o.getClass()) return false;
//        Pessoa pessoa = (Pessoa) o;
//        return Objects.equals(nome, pessoa.nome) &&
//                Objects.equals(rg, pessoa.rg);
//    }

    //Equals que compara apenas o RG do objeto
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Pessoa pessoa = (Pessoa) o;
        return Objects.equals(rg, pessoa.rg);
    }
}

File: app/src/main/java/intellij/class10/equalsoverloadingoverridetostring/exercise1/kotlin/Pessoa.kt
package intellij.class10.equalsoverloadingoverridetostring.exercise1.kotlin

data class Pessoa(var nome: String, var rg: Int) {

    //Equals padrão que compara o objeto inteiro
//    override fun equals(other: Any?): Boolean {
//        if (this === other) return true
//        if (javaClass != other?.javaClass) return false
//        other as Pessoa
//        if (nome != other.nome) return false
//        if (rg != other.rg) return false
//        return true
//    }

    //Equals que compara apenas o RG do objeto
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        other as Pessoa
        if (rg != other.rg) return false
        return true
    }
}